main.go

```diff
@@ -100,43 +106,68 @@ func (s *admissionWebhookServer) Review(in *admissionv1.AdmissionRequest) *admis
 
 func (s *admissionWebhookServer) unmarshal(in *admissionv1.AdmissionRequest) (p string, meta *v1.ObjectMeta, spec *corev1.PodSpec) {
 	var podSpec *corev1.PodSpec
-	var metaPtr *v1.ObjectMeta
+	var metaPtr, podMetaPtr *v1.ObjectMeta
 	var target interface{}
 	p = "/spec/template"
 	switch in.Kind.Kind {
-	case "Deployment":
+	case deploymentKind:
 		var deployment appsv1.Deployment
-		metaPtr = &deployment.Spec.Template.ObjectMeta
+		metaPtr = &deployment.ObjectMeta
+		podMetaPtr = &deployment.Spec.Template.ObjectMeta
 		podSpec = &deployment.Spec.Template.Spec
 		target = &deployment
-	case "Pod":
+	case podKind:
 		var pod corev1.Pod
 		p = ""
-		metaPtr = &pod.ObjectMeta
+		podMetaPtr = &pod.ObjectMeta
 		podSpec = &pod.Spec
 		target = &pod
 	case "DaemonSet":
 		var daemonSet appsv1.DaemonSet
-		metaPtr = &daemonSet.Spec.Template.ObjectMeta
+		metaPtr = &daemonSet.ObjectMeta
+		podMetaPtr = &daemonSet.Spec.Template.ObjectMeta
 		podSpec = &daemonSet.Spec.Template.Spec
 		target = &daemonSet
 	case "StatefulSet":
 		var statefulSet appsv1.StatefulSet
-		metaPtr = &statefulSet.Spec.Template.ObjectMeta
+		metaPtr = &statefulSet.ObjectMeta
+		podMetaPtr = &statefulSet.Spec.Template.ObjectMeta
 		podSpec = &statefulSet.Spec.Template.Spec
 		target = &statefulSet
+	case replicaSetKind:
+		var replicaSet appsv1.ReplicaSet
+		metaPtr = &replicaSet.ObjectMeta
+		podMetaPtr = &replicaSet.Spec.Template.ObjectMeta
+		podSpec = &replicaSet.Spec.Template.Spec
+		target = &replicaSet
+		defer func() {
+			s.logger.Info("Replicaset Defer method")
+			for _, o := range metaPtr.OwnerReferences {
+				if o.Kind == deploymentKind {
+					p, meta, spec = "", nil, nil
+				}
+			}
+		}()
```

Comment:
This is still looking overcomplicated.

I think the main problem here that we are doing things in `unmarshal` method that actually doesnt related to the operation.

Please move all  'post unmarshal' changes into separate method and use it right afiter unmarshal in the Review function.

Reply:
I've added changes, please take a look